base/range.jl

```diff
@@ -834,10 +834,10 @@ function getindex(r::AbstractUnitRange, s::AbstractUnitRange{T}) where {T<:Integ
             if first(s)
                 return r
             else
-                return range(r[1], length=0)
+                return range(first(r), length=0)
```

Comment:
If it's ok to use `last(r)` here, it's possible to use `range(ifelse(first(s), first(r), last(r)), length = Int(last(s)))` that condenses all the branches.

Reply:
Indeed this should be correct for all cases:
* `true:false`
* `true:true`
* `false:true`
* `false:false`
I have changed this